checksum.sh

```diff
@@ -1,18 +1,17 @@
 #!/bin/bash
 
 function checksum() {
+  local s
   s=$(curl -fsSL "$1")
   if ! command -v shasum >/dev/null
   then
     shasum() { sha1sum "$@"; }
```

Comment:
sha1sum doesn't support the `-a` option you always give below. You just shouldn't pass `-a`. I only used it when contriving an example (where I had to mock having something to download which has a hash already).

And you can just use a variable for the command (you don't need `eval` or anything like that to do such things).

```
local hasher
if command -v shasum >/dev/null ; then
  hasher=shasum
else
  hasher=sha1sum
fi
...
printf %s\\n "$s" | "$hasher" --check --status ...
```

Reply:
Updated. Great suggestion. Thanks!